#python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D0.sorted.bam,mapped/2D5N.sorted.bam,mapped/2D10N.sorted.bam,mapped/2D20R.sorted.bam,mapped/2D25.sorted.bam,mapped/2D5.sorted.bam,mapped/2D10.sorted.bam -b2  -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/es2_only -lite -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
#python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/8D0.sorted.bam,mapped/8D5N.sorted.bam,mapped/8D10N.sorted.bam,mapped/8D20R.sorted.bam,mapped/8D25.sorted.bam,mapped/8D5.sorted.bam,mapped/8D10.sorted.bam -b2  -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/es8_only -lite -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D0.sorted.bam,mapped/8D0.sorted.bam -b2 mapped/2D5N.sorted.bam,mapped/8D5N.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D0_vs_D5N  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D5N.sorted.bam,mapped/8D5N.sorted.bam -b2 mapped/2D10N.sorted.bam,mapped/8D10N.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D5N_vs_D10N  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D10N.sorted.bam,mapped/8D10N.sorted.bam -b2 mapped/2D20R.sorted.bam,mapped/8D20R.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D10N_vs_D20R  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D20R.sorted.bam,mapped/8D20R.sorted.bam -b2 mapped/2D25.sorted.bam,mapped/8D25.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D20R_vs_D25  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0

python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D10N.sorted.bam,mapped/8D10N.sorted.bam -b2 mapped/2D10.sorted.bam,mapped/8D10.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D10N_vs_D10  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0

python /Users/bioninja/program/rMATS.3.2.5/RNASeq-MATS.py -b1 mapped/2D5N.sorted.bam,mapped/8D5N.sorted.bam -b2 mapped/2D5.sorted.bam,mapped/8D5.sorted.bam -gtf /Users/bioninja/genome/gtf/gencode.v19.annotation.gtf -o alternative_splicing/D5N_vs_D5  -t paried -len 100 -a 1 -c 0.0001 -analysis U -novelSS 0
