#!bin/bash

qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D0_Aligned.out.bam,../STAR_out_for_all/8D0_Aligned.out.bam -b2 ../STAR_out_for_all/2D5N_Aligned.out.bam,../STAR_out_for_all/8D5N_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/0_vs_5N/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D5N_Aligned.out.bam,../STAR_out_for_all/8D5N_Aligned.out.bam -b2 ../STAR_out_for_all/2D20R_Aligned.out.bam,../STAR_out_for_all/8D10N_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/5N_vs_10N/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D5N_Aligned.out.bam,../STAR_out_for_all/8D5N_Aligned.out.bam -b2 ../STAR_out_for_all/2D5_Aligned.out.bam,../STAR_out_for_all/8D5_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/5N_vs_5/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D20R_Aligned.out.bam,../STAR_out_for_all/8D10N_Aligned.out.bam -b2 ../STAR_out_for_all/2D10_Aligned.out.bam,../STAR_out_for_all/8D10_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/10N_vs_10/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D20R_Aligned.out.bam,../STAR_out_for_all/8D10N_Aligned.out.bam -b2 ../STAR_out_for_all/2D10N_Aligned.out.bam,../STAR_out_for_all/8D20R_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/10N_vs_20R/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D10N_Aligned.out.bam,../STAR_out_for_all/8D20R_Aligned.out.bam -b2 ../STAR_out_for_all/2D25_Aligned.out.bam,../STAR_out_for_all/8D25_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/20R_vs_25/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D0_Aligned.out.bam,../STAR_out_for_all/8D0_Aligned.out.bam -b2 ../STAR_out_for_all/2D25_Aligned.out.bam,../STAR_out_for_all/8D25_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/0_vs_25/ -t paired -len 100
qsub -pe shared 4 -l h_data=4G,h_rt=30:00:00,highp ~/data/rMATs.sh python /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/RNASeq-MATS.py -b1 ../STAR_out_for_all/2D0_Aligned.out.bam,../STAR_out_for_all/8D0_Aligned.out.bam -b2 ../STAR_out_for_all/2D5_Aligned.out.bam,../STAR_out_for_all/8D5_Aligned.out.bam -gtf /u/home/z/zqfang/project-yxing/program/rMATS.3.0.9/gtf/Homo_sapiens.Ensembl.GRCh37.72.gtf -o /u/home/z/zqfang/project-yxing/STAR_neural_diff/STAR_rMATs/0_vs_5/ -t paired -len 100
